Source/WebCore/Modules/reporting/ReportingObserver.h

```diff
@@ -25,36 +25,42 @@
 
 #pragma once
 
-#include <optional>
-#include <wtf/Forward.h>
+#include "ContextDestructionObserver.h"
 #include <wtf/RefCounted.h>
-#include <wtf/Vector.h>
-#include <wtf/text/WTFString.h>
+#include <wtf/WeakPtr.h>
 
 namespace WebCore {
 
 class Report;
+class Reporting;
 class ReportingObserverCallback;
 class ScriptExecutionContext;
 
-class ReportingObserver final : public RefCounted<ReportingObserver> {
+class ReportingObserver final : public RefCounted<ReportingObserver>, public ContextDestructionObserver, public CanMakeWeakPtr<ReportingObserver>  {
```

Comment:
Do we really need a WeakPtr? I feel like CheckedPtr / CheckedRef is sufficient here.

Reply:
That's a good point. If I stored `Ref`'s to `ReportingObservers` in the `Reporting` object (owned by the `Document` or `WorkerOrWorkletGlobalScope`) it would prevent anything form being GC'd until they had been explicitly unregistered or the document was closed. I'll try that approach.